include/dlaf/common/assert.h

```diff
@@ -126,6 +126,8 @@ void silenceUnusedWarningFor(Args&&...) {}
 #define DLAF_STATIC_UNIMPLEMENTED(DummyType) \
   static_assert(sizeof(DummyType) == 0, "Not yet implemented!")
 
+#define DLAF_STATIC_FAIL(DummyType, Msg) static_assert(sizeof(DummyType) == 0, Msg)
```

Comment:
The name is not great, but it is a generic compile-time failure. It's definitely not `UNIMPLEMENTED` where I've used it. `UNREACHABLE` is close, but since it's compile-time it's not really right either. Hence I just used `FAIL`, but better suggestions are welcome.